src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,191 @@
+    TLmotor1.getPIDController().setP(TLP);
+    TLmotor1.getPIDController().setI(TLI);
+    TLmotor1.getPIDController().setD(TLD);
+    TLmotor2 = new CANSparkMax(2, MotorType.kBrushless);
+    TLmotor2.follow(TLmotor1);
+
+    BRmotor1 = new CANSparkMax(3, MotorType.kBrushless);
+    BRmotor1.setInverted(true);
+    BRmotor1.getPIDController().setP(BRP);
+    BRmotor1.getPIDController().setI(BRI);
+    BRmotor1.getPIDController().setD(BRD);
+    BRmotor2 = new CANSparkMax(4, MotorType.kBrushless);
+    BRmotor2.follow(BRmotor1);
+
+    feederMotor = new CANSparkMax(5, MotorType.kBrushless);
+
+    SmartDashboard.putNumber("shooterVelo", shooterVelo);
+    SmartDashboard.putNumber("feederVelo", feederVelo);
+
+    SmartDashboard.putNumber("TRShooterP", TLP);
+    SmartDashboard.putNumber("TRShooterI", TLI);
+    SmartDashboard.putNumber("TRShooterD", TLD);
+
+    SmartDashboard.putNumber("BRShooterP", BRP);
+    SmartDashboard.putNumber("BRShooterI", BRI);
+    SmartDashboard.putNumber("BRShooterD", BRD);
+  }
+
+  public Command RunMotors() {
+    return run(
+        () -> {
+          TLmotor1.getPIDController().setReference(shooterVelo, ControlType.kVelocity);
+          BRmotor1.getPIDController().setReference(shooterVelo, ControlType.kVelocity);
+        });
+  }
+
+  public Command StopMotors() {
+    return runOnce(
+        () -> {
+          TLmotor1.stopMotor();
+          BRmotor1.stopMotor();
+        });
+  }
+
+  public Command adjustSpeed(double testSpeedOffset) {
+    return run(
+      () -> {
+        shooterVelo+=testSpeedOffset;
+      }
+    );
+  }
+
+  public Command RunFeeder() {
+    return run(
+        () -> {
+          feederMotor.set(feederVelo);
+        });
+  }
+
+  public Command StopFeeder() {
+    return runOnce(
+        () -> {
+          feederMotor.stopMotor();
+        });
+  }
+
+  /**
+   * An example method querying a boolean state of the subsystem (for example, a digital sensor).
+   *
+   * @return value of some boolean subsystem state, such as a digital sensor.
+   */
+  public boolean exampleCondition() {
+    // Query some boolean state, such as a digital sensor.
+    return false;
+  }
+
+  @Override
+  public void periodic() {
+    shooterVelo = SmartDashboard.getNumber("shooterVelo", shooterVelo);
+    feederVelo = SmartDashboard.getNumber("feederVelo", feederVelo);
+    if(TLP != SmartDashboard.getNumber("TRShooterP", TLP)){
+      adjustTLP();
+    }
+    if(TLI != SmartDashboard.getNumber("TRShooterI", TLI)){
+      adjustTLI();
+    }
+    if(TLD != SmartDashboard.getNumber("TRShooterD", TLD)){
+      adjustTLD();
+    }
+
+    if(BRP != SmartDashboard.getNumber("BRShooterP", BRP)){
+        adjustBRP();
+    }
+    if(BRI != SmartDashboard.getNumber("BRShooterI", BRI)){
+        adjustBRI();
+    }
+    if(BRD != SmartDashboard.getNumber("BRShooterD", BRD)){
+        adjustBRD();
+    }
+
+    SmartDashboard.putNumber("shooterVelo", shooterVelo);
```

Comment:
Is this to allow changing the shooterVelo through the triggers?  
Seems like that means depending on the timing that shooterVelo can be overwritten by either trigger or shuffleboard.  Not sure.

Reply:
Reshik made it so that it could be changed through the triggers, but I think it would make more sense just to make it changeable through Shuffleboard